### PR TITLE
Fix: cannot set project TTL values to infinite

### LINKED ISSUE #523: Cannot set project TTL values to infinite
when using the latest terraform provider:
+ provider [registry.terraform.io/env0/env0](http://registry.terraform.io/env0/env0) v1.2.10

I am trying to create a project and specify the TTL values.  If I set the value to days, weeks, or months, it works fine.  However, I cannot set the max value to infinite as it's not an accepted variable/string, even though it's available in the UI.

Works:

```
resource "env0_project_policy" "example" {
  project_id                    = data.env0_project.customer-test.id
  default_ttl                   = "1-M"
  max_ttl                       = "1-M"
}
```

Fails:

```
resource "env0_project_policy" "example" {
  project_id                    = data.env0_project.customer-test.id
  default_ttl                   = "1-M"
  max_ttl                       = "Infinite"
}
```

Error:

```
│ Error: 'Infinite' must be one of: 6-h, 12-h, 1-d, 3-d, 1-w, 2-w, 1-M, inherit
│ 
│   with env0_project_policy.example,
│   on projects.tf line 14, in resource "env0_project_policy" "example":
│   14:   default_ttl                   = "Infinite"
```

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #523 

### Solution

Added "Infinite".
Fixed some bugs.
Updated acceptance tests.
Added an integration test.

### CODE REVIEW

File: client/project_policy.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package client
 
 type Policy struct {
-	Id                          string  `json:"id"`
+	Id                          string  `json:"id" tfschema:"-"`
```
Comment (TomerHeber) [inline client/project_policy.go:4]: The "Id" isn't used for resource_project_policy - instead the project id is used.
Ignoring this field.

File: client/project_policy.go [hunk 2]
```diff
@@ -28,8 +28,8 @@ type PolicyUpdatePayload struct {
 	SkipRedundantDeployments    bool   `json:"skipRedundantDeployments"`
 	RunPullRequestPlanDefault   bool   `json:"runPullRequestPlanDefault"`
 	ContinuousDeploymentDefault bool   `json:"continuousDeploymentDefault"`
-	MaxTtl                      string `json:"maxTtl"`
-	DefaultTtl                  string `json:"defaultTtl"`
+	MaxTtl                      string `json:"maxTtl,omitempty"`
```
Comment (TomerHeber) [inline client/project_policy.go:31]: If empty string. Don't send a json with an empty string. Instead don't send the field.
This should be equivalent to "null".

File: env0/resource_project_policy.go [hunk 3]
```diff
@@ -185,7 +189,9 @@ func resourcePolicyReset(ctx context.Context, d *schema.ResourceData, meta inter
 	}
 
 	d.SetId(projectId)
-	setPolicySchema(d, policy)
+	if err := writeResourceData(&policy, d); err != nil {
```
Comment (TomerHeber) [inline env0/resource_project_policy.go:192]: bug - writeResourceData was failing and was not caught. Adding "tfschema:"-"" fixes it.

**Comment by TomerHeber** [comment]
GitHub is having an issue... it's missing a commit.
Hopefully they fix it soon.

**Review by yaronya — APPROVED** [review]
LGTM
Just final question to get that straight - old behavior is still kept, is that correct? (i.e - when any TTL policy attribute is removed then it'd "reset" it)

**Comment by chrisgnoon** [comment]
Is this resolved now?

Best Regards

*Chris Noon*
Sales Engineer, env0

***@***.***

Schedule a meeting with me
<https://t.sidekickopen86.com/s3t/c/5/f18dQhb0S7kF8bp_dgVgZkbN59hl3kW7_k2841CX6NGW36PRXq1D8nbXW2PqZkh8wpHmCf197v5Y04?te=W3R5hFj4cm2zwW3F4Fph41QWmBW1JxwY51LwWh-W1N36fv49M9lff41WtSxV3&si=8000000024089145&pi=cc7087c4-e13c-4d58-b57f-534b24210641>

**Comment by TomerHeber** [comment]
> LGTM Just final question to get that straight - old behavior is still kept, is that correct? (i.e - when any TTL policy attribute is removed then it'd "reset" it)

@yaronya - I'm assuming that the "PUT" API call is not a "PATCH", and therefore should act as expected. I will test it just to be sure.

**Comment by TomerHeber** [comment]
> > LGTM Just final question to get that straight - old behavior is still kept, is that correct? (i.e - when any TTL policy attribute is removed then it'd "reset" it)
> 
> @yaronya - I'm assuming that the "PUT" API call is not a "PATCH", and therefore should act as expected. I will test it just to be sure.

@yaronya - made a quick test. (It's also documented). The default value is "inherit". Therefore, if removed it will switch to inherit as expected.

**Pull request merged**